### PR TITLE
Rephrase pull request template guidance to remove checklist

### PR DESCRIPTION
**Pull request by gonzalo-bulnes**
## Description

The check lists are picked up by GitHub as tasks to be completed,
which makes "radio"-style checklists misleading: some items
that GitHub identifies as tasks will never be completed!

While this is a matter of GitHub's interpretation, I think it is
useful to take advantage of the summary for evaluating how much
work is remaining on a PR.

Besides, the items were phrased as if tasks performed by
the pull request authors didn't need to be performed again
by the reviewer, which is not how we perform reviews.

## Test Plan    
     
- [ ] Confirm that the CI build is successful :green_apple: 
- [ ] Verify that this PR description seems useful to you :stuck_out_tongue_winking_eye: it uses the updated template! (Minus the emojis, I guess those are more of a personal taste.)
- [ ] Please proof-read me :grimacing: 

### Qubes OS environment    
     
If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes OS testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes OS is required. Please make sure you mention it in the **tests plan**.
     
### AppArmor profile    
     
If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please make sure you mention it in the **tests plan**.
     
### Database migrations    
     
If these changes modify the database schema, you should include a database migration. Testing that the migration applies cleanly on the `main` branch is necessary. Please make sure you mention it in the **tests plan**.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (eloquence) [inline .github/pull_request_template.md:17]: smol typo: tests plan -> test plan

**Comment by eloquence** [comment]
It's worth comparing the SD server template:
https://github.com/freedomofpress/securedrop/blob/develop/.github/PULL_REQUEST_TEMPLATE.md

Another way to avoid the "incomplete checklist" problem might be to avoid "choose one" checklists, and to instruct PR authors to delete non-applicable items & leave them unchecked if unsure. Then we'd want to always ensure every unchecked item is examined and checked or removed (if not applicable) prior to merge.

To be clear, I think either would be an improvement on the current format -- the incomplete checklists also trigger my mild OCD. ;-)

**Comment by gonzalo-bulnes** [comment]
(Amended the commit to fix the typos picked up by @eloquence :raised_hands:)

File: .github/pull_request_template.md
Comment (cfm) [inline .github/pull_request_template.md:21]: After b3cecdc85f9badc6924714053ef7851de9a9b8d1, I'd suggest:

```suggestion
If these changes modify the database schema, you should include a [self-contained](https://github.com/freedomofpress/securedrop-client#generating-and-running-database-migrations) database migration. Testing that the migration applies cleanly on the `main` branch is necessary. Please make sure you mention it in the **test plan**.
```

**Review by cfm — COMMENTED** [review]
@gonzalo-bulnes, I'll take you up on your invitation yesterday to provide informal feedback on this change.  :-)

I'm also bothered by the "incomplete checklists" problem.  But I think I favor (and I do tend to practice) @eloquence's suggestion in <https://github.com/freedomofpress/securedrop-client/pull/1521#issuecomment-1155875798>.

After rereading this pull request, my deeper concern is that this version of the pull-request template may actually be less helpful to community contributors, because they'll have to know (or, granted, ask) what to `mention in the test plan`, rather than having an explicit list of options to guide their actual work as well as their documentation of it.  I don't want to make the perfect the enemy of the good here, but I think I favor expanding and refining the checklists over eliminating them.  (And I'd be happy to collaborate on that effort.)

File: .github/pull_request_template.md
Comment (creviera) [inline .github/pull_request_template.md:13]: nit: "network (via the RPC service) traffic" -> "network traffic (via the RPC service)"

File: .github/pull_request_template.md
Comment (creviera) [inline .github/pull_request_template.md:7]: We might as well include the CI test job. Could just say "Confirm CI jobs are successful"

File: .github/pull_request_template.md
Comment (creviera) [inline .github/pull_request_template.md:17]: I like how this provides more details about why the apparmor profile would need to be updated than the checklist does.

**Review by creviera — CHANGES_REQUESTED** [review]
small changes requested

**Comment by creviera** [comment]
Also +1 to keeping a more useful checklist around with instructions about deleting non-applicable items. Another suggestion, is that there could be both a "PR owner checklist" and a "Reviewer checklist" (e.g. https://github.com/freedomofpress/securedrop-workstation-prod-rpm-packages-lfs/pull/31/files). Something to think about in terms of community PRs. The reviewer is probably a maintainer, so they'll be able to double-check the apparmor profile stuff and the like.

**Comment by gonzalo-bulnes** [comment]
I've included the wording suggestions from @cfm and @creviera.

I still find that checklists are less convenient than text for everyone (maintainers and occasional contributors), as they always require action (if only to remove items that are not relevant, and it seems like there are always items that won't be relevant). I'd prefer keeping with instructions, and using that as an opportunity to have a chat with external contributors if someone happens not to be sure how to write their test plan.

With my preferences stated, I'll leave it to you reviewers! :blue_heart:

**Comment by cfm** [comment]
Thanks, @gonzalo-bulnes.  Since this has process implications whatever one's preferences may be, I think I'll defer to @creviera as tech lead to give the final sign-off here.  :-)

**Comment by gonzalo-bulnes** [comment]
(rebased without changes to update the CI pipeline definitions)

**Comment by gonzalo-bulnes** [comment]
(rebased without changes to include new CI check)

**Review by creviera — CHANGES_REQUESTED** [review]
Nice improvement. Just a couple things to help newcomers. I think we should:
* link to or explain how to update the app armor profile
* add back something like "I would appreciate guidance" in the Qubes OS environment section because the PR owner might not be entirely sure if their say GUI code was considered "fine tuning"

**Comment by gonzalo-bulnes** [comment]
(Rebased without changes and added a commit to apply @creviera requested changes - I plan to squash both commits before merging.)

**Comment by gonzalo-bulnes** [comment]
There is no consensus about what the PR template is meant to achieve, so I'll close this PR until we get to that.

**Comment by eaon** [comment]
To add to the lack of consensus (for posterity/future conversations):

While I see (and kind of also feel) the intention with the checklists (_I totally feel the checked checklist dopamine_), I think their actual usefulness here is fairly limited. When I started here, they were kind of overwhelming, even though other open source projects I submitted PRs to have had similar approaches (but I worried about those checklists too).

I think the non-checklisted text is still useful as reminders for us (_did you remember this?_), while not being too "front-loady" for new contributors, so I like this better overall.

**Pull request closed without merging**
